execute/README.md

```diff
@@ -0,0 +1,456 @@
+The array values in a column are contained in contiguous data.
+When the chunk size gets larger, the memory allocator has to find a spot in memory that fits that large size.
+Larger chunk sizes are generally better for performance, but too large of buffer sizes put too much pressure on the memory allocator and garbage collector.
+The `table.BufferSize` constant is an agreed upon size to balance these concerns.
+At the moment, this value is the same as the buffer size that comes from the influxdb storage engine.
+If we find another buffer size works better in the future, we can change this one constant.
+
+### Prefer Column-Based Algorithms
+
+Column-based algorithms are generally faster than row-based algorithms.
+A column-based algorithm is one that lends itself to constructing each column individually instead of by row.
+Consider the following two examples:
+
+    b.Resize(10)
+    switch b := b.(type) {
+    case *array.FloatBuilder:
+        for i := 0; i < 10; i++ {
+            b.Append(rand.Float64())
+        }
+    case *array.IntBuilder:
+        for i := 0; i < 10; i++ {
+            b.Append(rand.Int64())
+        }
+    }
+
+    b.Resize(10)
+    for i := 0; i < 10; i++ {
+        switch b := b.(type) {
+        case *array.FloatBuilder:
+            b.Append(rand.Float64())
+        case *array.IntBuilder:
+            b.Append(rand.Int64())
+        }
+    }
+
+These are simple examples, but the first is column-based and the second is row-based.
+The row-based one spends a lot of time checking the type of the builder before appending the next value.
+This is a slow operation inside the for loop which itself is a hotspot for optimization.
+The first determines the type of the column first and then constructs the entire column using a specialized type.
+The first one only needs to pay the indirection cost of the interface once and can benefit from for loop optimizations.
+
+One practical example is `filter()`.
+The `filter()` function is row-based in that it evaluates each row independently to determine if it is filtered or not.
+A naive implementation would construct the builder for each row, resize it to the maximum capacity, and then append each value to each column whenever the row passed the filter.
+It would then finish by reallocating the arrays to the proper size.
+A faster method is to use a bitset to keep track of which rows are filtered and which ones remain.
+We can allocate a bitset that holds the rows in the current table chunk, run the filter on each row, and record whether it passes the filter or not.
+We can then use that bitset to construct each column independently of each other.
+
+### Utilize Slices
+
+Arrow arrays are immutable and have built in support for slicing data.
+A slice keeps the same reference to the underlying data, but limits the view of that data to a smaller section.
+Slices are most useful when the required data is already in order and only a section is needed.
```

Comment:
I'm not sure what you mean. I had a really hard time phrasing this so I could use some help making it more clear. What I'm trying to say here is that slices can be useful when the data naturally slices itself. Something like `limit()` might be a good example where you just need the data and you don't pay much of a cost. You don't release the unused memory as quickly, but sometimes that extra memory doesn't matter in the short term.

I just wanted to caution someone from using slices to get individual rows. Using slices for individual rows usually ends up making stuff slower.

Reply:
I would totally say something akin to what you've just said here then. You could even use the limit example to briefly show what you mean and leave out the word `section` entirely. Just a suggestion though. This might be more clear to other folks who have a look.

Reply:
Ok I updated this section.